crates/llms/src/responses.rs

```diff
@@ -31,24 +26,34 @@ use crate::chat::nsql::SqlGeneration;
 #[derive(Debug, Snafu)]
 #[snafu(visibility(pub))]
 pub enum Error {
-    #[snafu(display(
-        "Failed to load the model.\nAn error occurred: {source}\nReport a bug on GitHub: https://github.com/spiceai/spiceai/issues"
-    ))]
+    #[snafu(display("Failed to load the model: {source}"))]
     FailedToLoadModel {
         source: Box<dyn std::error::Error + Send + Sync>,
     },
-    #[snafu(display(
-        "Failed to run the model.\nAn error occurred: {source}\nReport a bug on GitHub: https://github.com/spiceai/spiceai/issues"
-    ))]
+    #[snafu(display("Failed to invoke the model: {source}"))]
     FailedToRunModel {
         source: Box<dyn std::error::Error + Send + Sync>,
     },
     #[snafu(display(
-        "Failed to check the status of the model. An error occurred: {source}. Verify the model configuration and try again."
+        "Failed to invoke the model: {source}. Verify the model configuration and try again."
     ))]
     HealthCheckError {
         source: Box<dyn std::error::Error + Send + Sync>,
     },
+    #[snafu(display("Failed to retrieve stream response from OpenAI: {source}."))]
+    StreamOpenAIError {
```

Comment:
This is a generic error struct - having specific errors for concrete implementations is a code smell